app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt

```diff
@@ -85,67 +101,65 @@ class HomeActivityTest {
     Intents.release()
   }
 
+  private fun setUpTestApplicationComponent() {
+    DaggerHomeActivityTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
   @Test
-  fun testHomeActivity_firstOpen_hasWelcomeString() {
-    launch(HomeActivity::class.java).use {
+  fun testHomeActivity_recyclerViewIndex0_withProfileId0_displayProfileName_profileNameDisplayedSuccessfully() {
+    launch<HomeActivity>(createHomeActivityIntent(0)).use {
       onView(
         atPositionOnView(
           R.id.home_recycler_view,
           0,
-          R.id.welcome_text_view
+          R.id.profile_name_textview
         )
-      ).check(matches(withText("Welcome to Oppia!")))
+      ).check(matches(withText("Sean!")))
     }
   }
 
   @Test
-  fun testHomeActivity_secondOpen_hasWelcomeBackString() {
-    simulateAppAlreadyOpened()
+  fun testHomeActivity_recyclerViewIndex0_getGreetingMessageBasedOnTime_goodMorningMessageSuccessful() {
+    launch<HomeActivity>(createHomeActivityIntent(0)).use {
```

Comment:
This test has nothing to do with recyclerview or the home activity. These test cases should go in utility module and in here you should check whether the text is getting displayed correctly or not.

Reply:
I have updated testcases.